### ISSUE TITLE
operators not supported by tflite: CAST, ExpandDims, FLOOR, Fill, Pow, RandomUniform, SPLIT, Stack, TensorFlowGreater, TensorFlowMaximum, TensorFlowMinimum, TensorFlowShape, TensorFlowSum, TensorFlowTile.

### ISSUE THREAD

**Issue by walkerlala** [tensorflow/tensorflow#20110]
This is a feature request issue.

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:

Yes

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:

CentOS 7

- **TensorFlow installed from (source or binary)**:

From `pip install tensorflow-gpu==1.6`

- **TensorFlow version (use command below)**:

tensorflow-gpu-1.6

- **Python version**:

Python 2.7.5
 
- **Bazel version (if compiling from source)**: None

- **GCC/Compiler version (if compiling from source)**: None

- **CUDA/cuDNN version**:  

CUDA: 9.0 

- **GPU model and memory**:

NVIDIA GTX 1080Ti, 11178MiB

- **Exact command to reproduce**:

```
toco --input_file=/tmp/mymodels/model_frozen.pb --output_file=/tmp/mymodels/converted_model.tflite --input_format=TENSORFLOW_GRAPHDEF --output_format=TFLITE --input_shape=1,320,320,3 --input_array=input_images --output_array=output_num_array --inference_type=FLOAT --input_data_type=FLOAT
```

### Describe the problem

I want to convert a trained model, which works on my GPU workstation, to a tflite model to work on Android phone. But I found that there are lots of operation not supported (see also the log below): **CAST, ExpandDims, FLOOR, Fill, Pow, RandomUniform, SPLIT, Stack, TensorFlowGreater, TensorFlowMaximum, TensorFlowMinimum, TensorFlowShape, TensorFlowSum, TensorFlowTile**. Do I need to provide customized operator for that (that is a lot of job...)? Or do newer version of toco support that? By the way, `toco` seems to not supported batch normalization either, see [this issue](https://github.com/tensorflow/tensorflow/issues/15336).

### Source code / logs

After running the command above, 

```
2018-06-19 10:17:59.783022: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1171] Converting unsupported operation: RandomUniform
2018-06-19 10:17:59.920127: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1171] Converting unsupported operation: Pow
2018-06-19 10:17:59.964930: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] Before Removing unused ops: 306 operators, 577 arrays (0 quantized)
2018-06-19 10:17:59.971329: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] Before general graph transformations: 306 operators, 577 arrays (0 quantized)
2018-06-19 10:18:02.355631: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] After general graph transformations pass 1: 216 operators, 476 arrays (0 quantized)
2018-06-19 10:18:02.360368: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] Before dequantization graph transformations: 216 operators, 476 arrays (0 quantized)
2018-06-19 10:18:02.364625: I tensorflow/contrib/lite/toco/allocate_transient_arrays.cc:311] Total transient array allocated size: 78643200 bytes, theoretical optimal value: 78643200 bytes.
2018-06-19 10:18:02.365858: F tensorflow/contrib/lite/toco/tflite/export.cc:304] Some of the operators in the model are not supported by the standard TensorFlow Lite runtime. If you have a custom implementation for them you can disable this error with --allow_custom_ops. Here is a list of operators for which you will need custom implementations: CAST, ExpandDims, FLOOR, Fill, Pow, RandomUniform, SPLIT, Stack, TensorFlowGreater, TensorFlowMaximum, TensorFlowMinimum, TensorFlowShape, TensorFlowSum, TensorFlowTile.
Aborted (core dumped)
```

**Comment by wenhui7909** [comment]
I met this error, and resolved by synchronizing tensorflow code.

**Comment by walkerlala** [comment]
@wenhui7909 You mean that lastest tensorflow code has solved that ?

**Comment by wenhui7909** [comment]
Yes,
Before, I also met this error , about can't find slice and some other op.
But now, those are ok.

Did u rebuild toco?

2018-06-22 10:43 GMT+08:00 Yubin <notifications@github.com>:

**Comment by bright1125** [comment]
@wenhui7909 I have met the same question,could you tell more about how to fix it?

**Comment by wenhui7909** [comment]
Update source code, and build toco as below:
bazel build tensorflow/contrib/lite/toco:toco

**Comment by bright1125** [comment]
@wenhui7909 I download the tensorflow r1.7 from github again,and bazel build again,but I didn't fix it.

**Comment by wenhui7909** [comment]
Try latest version or tensorflow 1.8,
and you also can check the op cc file is exist or not.
My error report that slice op is not supported, so check slice_op.cc.

**Comment by bright1125** [comment]
My error report that randomuniform,stack,tensorflowshape,floor
Tensorflow 1.8 don't support command line.I have try the 1.9 version.But the problem is still exist. 
My model code don't appear to have these ops ,if I check the op not exist,it seems complex to fix it.

**Comment by wenhui7909** [comment]
Oh, I don't know why I can see the op files and you can't.
such as, I can see floor.cc, and stack_ops.cc...

2018-07-03 17:36 GMT+08:00 KB <notifications@github.com>:

**Comment by anitha-v** [comment]
Would it be possible to provide your model ? It would help to understand if you could share what you are using those ops for.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrehentz: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by jdduke** [comment]
Slice should now be supported (as well as shape and floor). Would you mind running the converter again using the latest build (1.10, or head)? randomuniform is not yet supported, and we don't yet have an ETA for that op. If you share the model we might be able to advise some intermediate workarounds.

**Comment by tofulawrence** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
